lib/compress/huf_compress.c

```diff
@@ -795,6 +798,20 @@ HUF_compress_internal (void* dst, size_t dstSize,
                                            nbStreams, oldHufTable, bmi2);
     }
 
+    /* If uncompressible data is suspected, do a smaller sampling first */
+    if (suspectUncompressible && srcSize >= (SUSPECT_INCOMPRESSIBLE_SAMPLE_SIZE * SUSPECT_INCOMPRESSIBLE_SAMPLE_RATIO)) {
+        size_t largestTotal = 0;
+        {   unsigned maxSymbolValueBegin = maxSymbolValue;
+            CHECK_V_F(largestBegin, HIST_count_wksp (table->count, &maxSymbolValueBegin, (const BYTE*)src, SUSPECT_INCOMPRESSIBLE_SAMPLE_SIZE, workSpace_align4, wkspSize) );
```

Comment:
possible performance tip : 
presuming that input is not compressible,
the histogram may not need the 4 parallel counts designed to reduce the impact of successive store/load at the same position.

In which case, `HIST_count_simple()` should  count as fast as `HIST_count_wksp()`. The advantage though is that it wouldn't need  any `workspace`, and could possibly end up being slightly faster thanks to lower initialization costs.

Reply:
Q: Have you re-checked performance after this change ?